server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -98,8 +128,14 @@ public void parse(ParseContext context) throws IOException {
 
     @Override
     protected void parseCreateField(ParseContext context) throws IOException {
+        var fieldType = fieldType().hasDocValues() ? Defaults.FIELD_TYPE : Defaults.LEGACY_FIELD_TYPE;
         for (String field : context.getIgnoredFields()) {
-            context.doc().add(new Field(NAME, field, Defaults.FIELD_TYPE));
+            if (fieldType().hasDocValues()) {
+                final BytesRef binaryValue = new BytesRef(field);
+                context.doc().add(new SortedSetDocValuesField(fieldType().name(), binaryValue));
+            } else {
+                context.doc().add(new Field(NAME, field, fieldType));
```

Comment:
Sorry I wasn't clear, we should always add the `Field` version, it makes the `_ignored` field searchable. 
So you can do  something like:
```yml
if (fieldType().hasDocValues()) {
  final BytesRef binaryValue = new BytesRef(field);
  context.doc().add(new SortedSetDocValuesField(fieldType().name(), binaryValue));
}
context.doc().add(new Field(NAME, field, fieldType));
```

Reply:
I tried this, but then I get duplicate values for "_ignored" in the source

Reply:
You mean you have duplicate `_ignored` values in the response ? Could it be because you set stored to true in the new `FIELD_TYPE` ? It should be the other way around, the new field type shouldn't store the value as I commented above.

Reply:
You are correct it was because of the inverse stored.
Once I flipped them, it works as expected